### PR TITLE
fix X-Forwarded-For header handling

### PR DESCRIPTION
**Pull request by pchojnacki**
/cc: @pwojcik86  @sqreek

### CODE REVIEW

**Comment by sqreek** [comment]
LG

File: nginx/production/api-gateway.conf [hunk 1]
```diff
@@ -2,7 +2,7 @@ server {
   listen 8100 default_server;
   server_name "" [::]:8100;
   resolver 127.0.0.1:8600;
-  proxy_set_header X-Forwarded-For $remote_addr;
+  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```
Comment (drsnyder) [inline nginx/production/api-gateway.conf]: Where is this being set? The `$remote_addr` is set by the `real_ip` module to be the IP address that we get from Fastly.

**Comment by drsnyder** [comment]
Sorry to nit pick. Please add a description to the PR so that others know what it's about.

I don't think this is the right way to send the client IP to the service. The `$remote_addr` is set to what we get from Fastly via `Fastly-Client-IP` [here](https://github.com/Wikia/chef-repo/blob/d6c9fa1b61e779845e04998c906da2ef0c21c8c5/cookbooks/_nginx/templates/default/real-ip-module.conf.erb). That header gives us _the_ client IP and not the chain of IPs. It allows us to clarify the logs and it enables effective rate limiting.

**Comment by pchojnacki** [comment]
Sorry I wasn't aware about our use of real_ip module. This PR was made as a somewhat shot in the dark to fix the problem with user-registration connecting to mediawiki through fastly which sets its own XFF headers for mediawiki.
Overall I think we'll keep $proxy_add_x_forwarded_for in api gateway but we'll standardize this header contents so it will match more closely to how it is handled in mediawiki. As for user-registration it will either have to parse XFF or we'll make it expect X-Client-Ip which is header that Wladek proposed in his guideline.
I've created epic https://wikia-inc.atlassian.net/browse/SER-191 to track this change.

**Comment by drsnyder** [comment]
User-registration should NOT connect to mediawiki through fastly. It uses `prod.icache.service.consul` which I believe is an internal varnish. If there is an issue with the client header being set in the proxy through `prod.icache.service.consul` I'm not sure this is the right place to make that fix since this change *could* break downstream clients that are expecting *one and only one* client IP.

I just want to be careful that we aren't making a change to satisfy MW while making it harder for all other services behind the gateway to extract the client IP.

**Comment by pchojnacki** [comment]
Until recently User-registration was connecting through Fastly IIRC but that should already be resolved. https://github.com/Wikia/pandora/commit/881ea23730d599badc6634e38005224230927c4c

problem with X-Forwarded-For header is that we have disparity in how we treat it in services and in MW.
I believe As long as we'll provide viable alternative for services via x-client-ip then it shouldn't be aproblem to standardize xff header for all apps.

**Pull request merged**